### ISSUE TITLE
Json plugin reports are not consistently same

### ISSUE THREAD

**Issue by manavghaigithub** [getgauge-contrib/json-report#14]
Json plugin, for the same test, sometimes shows steps details and sometime shows details only at spec level.

I ran my test twice and both the times the report is different. In one report i see that all the step details are present but the second report does not has any step detail. The behavior is inconsistent.

Attached are the files for your reference.
[Json Results.zip](https://github.com/getgauge-contrib/json-report/files/3973722/Json.Results.zip)

**Comment by manavghaigithub** [comment]
@sriv @mahendrakariya @renjithgr @nehashri @apoorvam @shubhamsc @BugDiver 
Hello - Can somebody please look into this and advise? Thanks in advance !!! 
Thanks Manav

**Comment by sriv** [comment]
@manavghaigithub - do you have a sample project for us to check? 

Also, do you see any difference in other reports (console output/html-report/xml-report) ?

**Comment by manavghaigithub** [comment]
@sriv I am in the network so cannot share project but i did share actual Json reports (from project execution) when i opened ticket. In the reports you can see that one report has detailed information upto Step level and another report has details only until Spec level.

This behavior is intermittent and we are trying to understand what is causing it. We did not make any changes before executing project when these reports were generated.

**Comment by sriv** [comment]
> I am in the network so cannot share project but i did share actual Json reports 

Yes, I noticed those too. But in order to find a cause (and confirm a fix), we'll need a way to reproduce the issue. 

> This behavior is intermittent and we are trying to understand what is causing it.

Oh boy, that's never good. I request that you keep an eye for any pattern that can give us some hint on what could be the source for this problem. 

Both Gauge and JSON-report plugin are involved in generating this output. Gauge gives out the raw output, whereas the plugin transforms it to JSON. I am trying to narrow the problem to one of these, hence the request for a sample project to reproduce the issue. I hope this is reasonable.
